airflow/jobs.py

```diff
@@ -407,16 +407,18 @@ def schedule_dag(self, dag):
                             # add % as a wildcard for the like query
                             DagRun.run_id.like(DagRun.ID_PREFIX+'%')))
             last_scheduled_run = qry.scalar()
-            next_run_date = None
-            if dag.schedule_interval == '@once' and not last_scheduled_run:
-                next_run_date = datetime.now()
-            elif not last_scheduled_run:
+            
+            # don't schedule @once again
+            if dag.schedule_interval == '@once' and last_scheduled_run:
+                return None
+
+            if not last_scheduled_run:
                 # First run
                 TI = models.TaskInstance
                 latest_run = (
                     session.query(func.max(TI.execution_date))
-                    .filter_by(dag_id=dag.dag_id)
-                    .scalar()
+                        .filter_by(dag_id=dag.dag_id)
+                        .scalar()
                 )
                 if latest_run:
```

Comment:
I'm a little confused about when this would be `True`? The query almost the same as `qry`, above.